src/NuGet.Core/NuGet.ProjectModel/ProjectRestoreMetadata.cs

```diff
@@ -231,6 +249,40 @@ public virtual ProjectRestoreMetadata Clone()
             return clone;
         }
 
+        public static NuGetVersion GetSdkAnalysisLevel(string sdkAnalysisLevel)
```

Comment:
This isn't used everywhere where it could be (example, not used in the nomination codepath).

It does promote code reuse, but we should also make sure the logic that we're adding is really worth that.

That being said, we do have a precedent, which is putting the helper methods in https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/MSBuildRestoreUtility.cs.